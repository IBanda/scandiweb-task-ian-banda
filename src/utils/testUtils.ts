import { waitForElementToBeRemoved, screen } from '@testing-library/react';

export const load = async () => {
     await waitForElementToBeRemoved(screen.getByTestId('loader'));
};
