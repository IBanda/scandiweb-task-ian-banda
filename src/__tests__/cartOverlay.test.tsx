import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TestApp from '../utils/test_utils/TestApp';
import { load, renderWithRouter } from '../utils/test_utils/testUtils';

beforeEach(async () => {
     renderWithRouter(<TestApp />);
     await load();
});

test('Empty cart should show message', async () => {
     userEvent.click(screen.getByTestId('cart_overlay_btn'));

     expect(screen.getByText(/cart is empty/i)).toBeInTheDocument();
});

test('Added product should show in cart overlay', async () => {
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('cart_overlay_btn'));

     expect(screen.getByTestId('cart_list_item')).toBeInTheDocument();
     expect(screen.getByTestId('cart_list')).toMatchSnapshot();
});

test('Car count should show total quantity', () => {
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('cart_overlay_btn'));

     expect(screen.getByText(/2 items/i)).toBeInTheDocument();
});

test('Should change product quantity in overlay', () => {
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('cart_overlay_btn'));

     userEvent.click(screen.getByTestId('btn_inc'));

     expect(screen.getByTestId('quantity')).toHaveTextContent('2');
     expect(screen.getByTestId('indicator')).toHaveTextContent('2');
});

test('A quantity of 0 should remove product from cart', () => {
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('cart_overlay_btn'));

     userEvent.click(screen.getByTestId('btn_dec'));

     expect(screen.getByText(/cart is empty/i)).toBeInTheDocument();
});

test('First attribute should be selected', () => {
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('cart_overlay_btn'));

     expect(screen.getByTestId(/size-0/i)).toHaveClass('selected');
});

test('View bag should navigate to cart page', () => {
     userEvent.click(screen.getByTestId('add_to_cart'));
     userEvent.click(screen.getByTestId('cart_overlay_btn'));
     userEvent.click(screen.getByText(/view bag/i));

     expect(screen.getByText(/cart/i)).toBeInTheDocument();
});
