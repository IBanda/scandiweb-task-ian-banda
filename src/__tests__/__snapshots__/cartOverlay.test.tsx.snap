// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Added product should show in cart overlay 1`] = `
<ul
  class="sc-hKwDye cJMKyJ cart_list"
  data-testid="cart_list"
>
  <li
    class="cart_item"
    data-testid="cart_list_item"
  >
    <div
      class="cart_left"
    >
      <h4
        class="brand"
      >
        <a
          href="/product/huarache-x-stussy-le"
        >
          Nike x Stussy
        </a>
      </h4>
      <h5
        class="name"
        data-testid="cart_list-product_name"
      >
        <a
          href="/product/huarache-x-stussy-le"
        >
          Nike Air Huarache Le
        </a>
      </h5>
      <h6
        class="price"
      >
        $
         
        144.69
      </h6>
      <div
        class="sc-gsDKAQ kBWjMo attr_box_container"
      >
        <div
          class="attr_box 
                                                       selected
                                                       "
          data-testid="Size-0"
        >
          <input
            class="attr_btn"
            name="Size"
            type="radio"
            value="40"
          />
          <span
            class="attr_text"
          >
            40
          </span>
        </div>
        <div
          class="attr_box 
                                                       
                                                       "
          data-testid="Size-1"
        >
          <input
            class="attr_btn"
            name="Size"
            type="radio"
            value="41"
          />
          <span
            class="attr_text"
          >
            41
          </span>
        </div>
        <div
          class="attr_box 
                                                       
                                                       "
          data-testid="Size-2"
        >
          <input
            class="attr_btn"
            name="Size"
            type="radio"
            value="42"
          />
          <span
            class="attr_text"
          >
            42
          </span>
        </div>
        <div
          class="attr_box 
                                                       
                                                       "
          data-testid="Size-3"
        >
          <input
            class="attr_btn"
            name="Size"
            type="radio"
            value="43"
          />
          <span
            class="attr_text"
          >
            43
          </span>
        </div>
      </div>
    </div>
    <div
      class="cart_right"
    >
      <div
        class="sc-dkPtRN dzPSCc quantity"
      >
        <button
          data-testid="btn_inc"
        >
          ＋
        </button>
        <span
          class="quantity_text"
          data-testid="quantity"
        >
          1
        </span>
        <button
          class="dd"
          data-testid="btn_dec"
        >
          —
        </button>
      </div>
      <img
        alt="product"
        class="cart_img"
        src="https://cdn.shopify.com/s/files/1/0087/6193/3920/products/DD1381200_DEOA_2_720x.jpg?v=1612816087"
      />
    </div>
  </li>
</ul>
`;
